src/core/id-headers.js

```diff
@@ -43,9 +43,13 @@ export function run(conf) {
       h.closest(".appendix") ? "Appendix" : "Section",
       h.querySelector(":scope > bdi.secno")
     );
-    h.insertAdjacentElement(
-      "afterend",
-      html`<a href="${`#${id}`}" class="self-link" aria-label="${label}"></a>`
-    );
+    const wrapper = html`<div class="header-wrapper"></div>`;
+    h.replaceWith(wrapper);
+    const selfLink = html`<a
+      href="#${id}"
+      class="self-link"
+      aria-label="${label}"
+    ></a>`;
+    wrapper.append(h, selfLink);
```

Comment:
Can we do something like following?

```js
html`<div class="header-wrapper>${h}<a class="self-link">...</a></div>`
```

Reply:
Not sure. I’m ok with current approach to as it’s clear that it’s replacing the header.